packages/fiber/src/core/stages.ts

```diff
@@ -0,0 +1,149 @@
+import { MutableRefObject } from 'react'
+import { StoreApi, UseBoundStore } from 'zustand'
+import { RootState } from './store'
+
+export interface UpdateCallback {
+  (state: RootState, delta: number, frame?: THREE.XRFrame | undefined): void
+}
+
+export type UpdateCallbackRef = MutableRefObject<UpdateCallback>
+type Store = UseBoundStore<RootState, StoreApi<RootState>>
+export type UpdateSubscription = { ref: UpdateCallbackRef; store: Store }
+
+export type FixedStageOptions = { fixedStep?: number; maxSubsteps?: number }
+export type FixedStageProps = { fixedStep: number; maxSubsteps: number; accumulator: number; alpha: number }
+
+/**
+ * Class representing a stage that updates every frame.
+ * Stages are used to build a lifecycle of effects for an app's frameloop.
+ * @param {string} name - Name of the stage.
+ */
+export class Stage {
+  name: string
+  subscribers: UpdateSubscription[]
+
+  constructor(name: string) {
```

Comment:
After the recent update, do we still need to pass stage names to the constructors?